<br>
<h1 style = "font-size:40px; font-family:cursive ; font-weight : normal; color : #C70039; text-align: center; border-radius: 10px 100px;">IPL Dataset Analysis(2008-2020)</h1>
<br>



<h1 style = "font-family: Snell Roundhand;font-weight: 500;font-size: 30px; font-style: normal; background-color: #FFFB30; color :black;">Importing libraries:</h1>

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import plotly
import plotly.express as px
import plotly.graph_objs as go
from plotly.offline import init_notebook_mode, plot, iplot
from plotly import tools

import warnings
warnings.filterwarnings('ignore')

<h1 style = "font-family: Snell Roundhand;font-weight: 500;font-size: 30px; font-style: normal; background-color: #FFFB30; color :black;"> Loading match and deliveries dataset:</h1>

In [2]:
deliveries_data = pd.read_csv('/content/IPL Ball-by-Ball 2008-2020.csv')
match_data = pd.read_csv('/content/IPL Matches 2008-2020 (1).csv')
print("Data ready for exploration")

Data ready for exploration


<h1 style = "font-family: Snell Roundhand;font-weight: 500;font-size: 30px; font-style: normal; background-color: #FFFB30; color :black;"> Basic data exploration:</h1>


In [3]:
match_data.head()

id        city        date  ... method    umpire1         umpire2
0  335982   Bangalore  2008-04-18  ...    NaN  Asad Rauf     RE Koertzen
1  335983  Chandigarh  2008-04-19  ...    NaN  MR Benson      SL Shastri
2  335984       Delhi  2008-04-19  ...    NaN  Aleem Dar  GA Pratapkumar
3  335985      Mumbai  2008-04-20  ...    NaN   SJ Davis       DJ Harper
4  335986     Kolkata  2008-04-20  ...    NaN  BF Bowden     K Hariharan

[5 rows x 17 columns]

In [4]:
deliveries_data.head()

id  inning  ...           batting_team                 bowling_team
0  335982       1  ...  Kolkata Knight Riders  Royal Challengers Bangalore
1  335982       1  ...  Kolkata Knight Riders  Royal Challengers Bangalore
2  335982       1  ...  Kolkata Knight Riders  Royal Challengers Bangalore
3  335982       1  ...  Kolkata Knight Riders  Royal Challengers Bangalore
4  335982       1  ...  Kolkata Knight Riders  Royal Challengers Bangalore

[5 rows x 18 columns]

In [5]:
match_data.isnull().sum()

id                   0
city                13
date                 0
player_of_match      4
venue                0
neutral_venue        0
team1                0
team2                0
toss_winner          0
toss_decision        0
winner               4
result               4
result_margin       17
eliminator           4
method             797
umpire1              0
umpire2              0
dtype: int64

In [6]:
match_data.shape

(816, 17)

In [7]:
match_data.columns

Index(['id', 'city', 'date', 'player_of_match', 'venue', 'neutral_venue',
       'team1', 'team2', 'toss_winner', 'toss_decision', 'winner', 'result',
       'result_margin', 'eliminator', 'method', 'umpire1', 'umpire2'],
      dtype='object')

In [8]:
print('Total Matches Played:',match_data.shape[0])
print(' \n Venues Played At:',match_data['city'].unique())     
print(' \n Teams :',match_data['team1'].unique())

Total Matches Played: 816
 
 Venues Played At: ['Bangalore' 'Chandigarh' 'Delhi' 'Mumbai' 'Kolkata' 'Jaipur' 'Hyderabad'
 'Chennai' 'Cape Town' 'Port Elizabeth' 'Durban' 'Centurion' 'East London'
 'Johannesburg' 'Kimberley' 'Bloemfontein' 'Ahmedabad' 'Cuttack' 'Nagpur'
 'Dharamsala' 'Kochi' 'Indore' 'Visakhapatnam' 'Pune' 'Raipur' 'Ranchi'
 'Abu Dhabi' nan 'Rajkot' 'Kanpur' 'Bengaluru' 'Dubai' 'Sharjah']
 
 Teams : ['Royal Challengers Bangalore' 'Kings XI Punjab' 'Delhi Daredevils'
 'Mumbai Indians' 'Kolkata Knight Riders' 'Rajasthan Royals'
 'Deccan Chargers' 'Chennai Super Kings' 'Kochi Tuskers Kerala'
 'Pune Warriors' 'Sunrisers Hyderabad' 'Gujarat Lions'
 'Rising Pune Supergiants' 'Rising Pune Supergiant' 'Delhi Capitals']


<a id="1">
<h1 style = "font-family: Snell Roundhand;font-weight: 500;font-size: 30px; font-style: normal; background-color: #FFFB30; color :black;"> 1) Number of matches played in various seasons :</h1>


In [9]:
match_data['Season'] = pd.DatetimeIndex(match_data['date']).year
match_data.head()

id        city        date  ...    umpire1         umpire2  Season
0  335982   Bangalore  2008-04-18  ...  Asad Rauf     RE Koertzen    2008
1  335983  Chandigarh  2008-04-19  ...  MR Benson      SL Shastri    2008
2  335984       Delhi  2008-04-19  ...  Aleem Dar  GA Pratapkumar    2008
3  335985      Mumbai  2008-04-20  ...   SJ Davis       DJ Harper    2008
4  335986     Kolkata  2008-04-20  ...  BF Bowden     K Hariharan    2008

[5 rows x 18 columns]

In [10]:
match_per_season=match_data.groupby(['Season'])['id'].count().reset_index().rename(columns={'id':'matches'})
match_per_season.style.background_gradient(cmap='PuBu')

In [11]:
colors = ['turquoise',] * 13
colors[5] = 'crimson'

fig=px.bar(data_frame=match_per_season,x=match_per_season.Season,y=match_per_season.matches,labels=dict(x="Season",y="Count"),)
fig.update_layout(title="Number of matches played in different seasons ",
                  titlefont={'size': 26},template='simple_white'     
                  )
fig.update_traces(marker_line_color='black',
                  marker_line_width=2.5, opacity=1,marker_color=colors)
fig.show()

<div class="alert alert-block alert-info"><p style='color:black;'> 📌Each season, almost 60 matches were played. However, we see a spike in the number of matches from 2011 to 2013. This is because two new franchises, the Pune Warriors and Kochi Tuskers Kerala, were introduced, increasing the number of teams to 10.</p>
</div>

<small><a href='#0'>Go to top</a></small>  

<a id="2">
<h1 style = "font-family: Snell Roundhand;font-weight: 500;font-size: 30px; font-style: normal; background-color: #FFFB30; color :black;"> 2) Total number of runs scored across seasons :</h1>


In [12]:
season_data=match_data[['id','Season']].merge(deliveries_data, left_on = 'id', right_on = 'id', how = 'left').drop('id', axis = 1)
season_data.head()

Season  inning  ...           batting_team                 bowling_team
0    2008       1  ...  Kolkata Knight Riders  Royal Challengers Bangalore
1    2008       1  ...  Kolkata Knight Riders  Royal Challengers Bangalore
2    2008       1  ...  Kolkata Knight Riders  Royal Challengers Bangalore
3    2008       1  ...  Kolkata Knight Riders  Royal Challengers Bangalore
4    2008       1  ...  Kolkata Knight Riders  Royal Challengers Bangalore

[5 rows x 18 columns]

In [13]:
# Season
season=season_data.groupby(['Season'])['total_runs'].sum().reset_index()
p=season.set_index('Season')
fig = px.line(p, x=p.index, y="total_runs")
fig.update_layout(title="Total Runs Across the Seasons ",
                  titlefont={'size': 26},template='simple_white'     
                  )
fig.show()

</div>
<div class="alert alert-block alert-info"><p style='color:black;'> 📌<b>Season 2013</b> was the highest scoring season (22,541 runs), followed by 2012 (22,453 runs)<br>
<b>Season 2009</b> was the lowest scoring season (16,320 runs).</p>
</div>

<small><a href='#0'>Go to top</a></small>  

<a id="3">
<h1 style = "font-family: Snell Roundhand;font-weight: 500;font-size: 30px; font-style: normal; background-color: #FFFB30; color :black;"> 3) Runs scored per match across seasons :</h1>


In [14]:
runs_per_season=pd.concat([match_per_season,season.iloc[:,1]],axis=1)
runs_per_season['Runs scored per match']=runs_per_season['total_runs']/runs_per_season['matches']
runs_per_season.set_index('Season',inplace=True)
runs_per_season.style.background_gradient(cmap='PuBu',subset=['Runs scored per match'])

In [15]:
fig = px.line(runs_per_season, x=runs_per_season.index, y="Runs scored per match")
fig.update_layout(title="Runs scored per match across seasons",
                  titlefont={'size': 26},template='simple_white'     
                  )
fig.show()


<div class="alert alert-block alert-info"><p style='color:black;'> 📌In <b>season 2018</b>, runs scored per match was 331.683333 which was highest among others.<br> 
In <b>season 2009</b>, runs scored per match was 286.315789 which was lowest till now.</p> </div>

<small><a href='#0'>Go to top</a></small>  

<a id="4">
<h1 style = "font-family: Snell Roundhand;font-weight: 500;font-size: 30px; font-style: normal; background-color: #FFFB30; color :black"> 4) Count of matches by umpires :</h1>



In [16]:
ump=pd.concat([match_data['umpire1'],match_data['umpire2']])
ump=ump.value_counts()
umps=ump.to_frame().reset_index()
ump.head(10)

S Ravi             121
HDPK Dharmasena     94
AK Chaudhary        87
C Shamshuddin       82
M Erasmus           65
Nitin Menon         57
CK Nandan           57
SJA Taufel          55
Asad Rauf           51
VA Kulkarni         50
dtype: int64

<div class="alert alert-block alert-info"><p style='color:black;'> 📌<b>S Ravi</b> has umpired in 121 matches, followed by Dharmasena who has umpired in 94 matches.
</p>
</div>

<small><a href='#0'>Go to top</a></small>  

<a id="5">
<h1 style = "font-family: Snell Roundhand;font-weight: 500;font-size: 30px; font-style: normal; background-color: #FFFB30; color :black;"> 5) Number of tosses won by teams :</h1>




In [17]:
match_data['toss_winner'].value_counts()

Mumbai Indians                 106
Kolkata Knight Riders           98
Chennai Super Kings             97
Rajasthan Royals                87
Royal Challengers Bangalore     87
Kings XI Punjab                 85
Delhi Daredevils                80
Sunrisers Hyderabad             57
Deccan Chargers                 43
Pune Warriors                   20
Delhi Capitals                  20
Gujarat Lions                   15
Kochi Tuskers Kerala             8
Rising Pune Supergiants          7
Rising Pune Supergiant           6
Name: toss_winner, dtype: int64

In [18]:
toss=match_data['toss_winner'].value_counts()
colors = ['turquoise',] * 15
colors[0] = 'crimson'
fig=px.bar( y=toss,x=toss.index,labels=dict(x="Season",y="Count"),)
fig.update_layout(title="No. of tosses won by each team",
                  titlefont={'size': 26},template='simple_white'     
                  )
fig.update_traces(marker_line_color='black',
                  marker_line_width=2.5, opacity=1,marker_color=colors)
fig.show()

<div class="alert alert-block alert-info"><p style='color:black;'> 📌<b>Mumbai Indians</b> have won the most tosses, followed by Kolkata Knight Riders.
</p>
</div>

<small><a href='#0'>Go to top</a></small>  

<a id="6">
<h1 style = "font-family: Snell Roundhand;font-weight: 500;font-size: 30px; font-style: normal; background-color: #FFFB30; color :black;"> 6) Decision made after winning the toss :</h1>

In [19]:
temp_series = match_data.toss_decision.value_counts()
labels = (np.array(temp_series.index))
values = (np.array((temp_series / temp_series.sum())*100))
colors = ['turquoise', 'crimson']
fig = go.Figure(data=[go.Pie(labels=labels,
                             values=values,hole=.3)])
fig.update_traces(hoverinfo='label+percent', textinfo='label+percent', textfont_size=20,
                  marker=dict(colors=colors, line=dict(color='#000000', width=3)))
fig.update_layout(title="Toss decision percentage",
                  titlefont={'size': 30},
                  )
fig.show()

<div class="alert alert-block alert-info"><p style='color:black;'> 📌After winning the toss, team tends to field first<br>
</p>
</div>

<small><a href='#0'>Go to top</a></small>  

<a id="7">
<h1 style = "font-family: Snell Roundhand;font-weight: 500;font-size: 30px; font-style: normal; background-color: #FFFB30; color :black;"> 7) Toss decision across seasons :</h1>



In [20]:
match_data.head()

id        city        date  ...    umpire1         umpire2  Season
0  335982   Bangalore  2008-04-18  ...  Asad Rauf     RE Koertzen    2008
1  335983  Chandigarh  2008-04-19  ...  MR Benson      SL Shastri    2008
2  335984       Delhi  2008-04-19  ...  Aleem Dar  GA Pratapkumar    2008
3  335985      Mumbai  2008-04-20  ...   SJ Davis       DJ Harper    2008
4  335986     Kolkata  2008-04-20  ...  BF Bowden     K Hariharan    2008

[5 rows x 18 columns]

In [21]:
fig=px.histogram(data_frame=match_data,x='Season',color='toss_decision',color_discrete_sequence=colors,barmode='group')
fig.update_layout(title="Toss decision in different seasons",
                  titlefont={'size': 26},template='simple_white'     
                  )
fig.update_traces(marker_line_color='black',
                  marker_line_width=2.5, opacity=1)
fig.show()

<div class="alert alert-block alert-info"><p style='color:black;'> 📌Most of the times, teams decide to feild first except in season 2009,2010,2013 where teams decided to bat first mostly.<br>
Since 2014, teams have overwhelmingly chosen to bat second. Especially since 2016, teams have chosen to field for more than 80% of the times except in season 2020.<br>
</p>
</div>

<small><a href='#0'>Go to top</a></small>  

<a id="8">
<h1 style = "font-family: Snell Roundhand;font-weight: 500;font-size: 30px; font-style: normal; background-color: #FFFB30; color :black;"> 8)Winning toss implies winning game ?</h1>


In [22]:
match_data['toss_win_game_win'] = np.where((match_data.toss_winner == match_data.winner),'Yes','No')
match_data.head()

id        city        date  ...         umpire2 Season  toss_win_game_win
0  335982   Bangalore  2008-04-18  ...     RE Koertzen   2008                 No
1  335983  Chandigarh  2008-04-19  ...      SL Shastri   2008                Yes
2  335984       Delhi  2008-04-19  ...  GA Pratapkumar   2008                 No
3  335985      Mumbai  2008-04-20  ...       DJ Harper   2008                 No
4  335986     Kolkata  2008-04-20  ...     K Hariharan   2008                 No

[5 rows x 19 columns]

In [23]:
labels =["Yes",'No']
values = match_data['toss_win_game_win'].value_counts()
colors = ['turquoise', 'crimson']
fig = go.Figure(data=[go.Pie(labels=labels,
                             values=values,hole=.3)])
fig.update_traces(hoverinfo='label+percent', textinfo='label+percent', textfont_size=20,
                  marker=dict(colors=colors, line=dict(color='#000000', width=3)))
fig.update_layout(title="Winning toss implies winning macthes?",
                  titlefont={'size': 30},
                  )
fig.show()

<div class="alert alert-block alert-info"><p style='color:black;'> 📌Though winning toss gives you an advantage <b>but it doesn't significantly implies that winning the toss helps in winning the game.</b>
</p>
</div>

<a id="9">
<h1 style = "font-family: Snell Roundhand;font-weight: 500;font-size: 30px; font-style: normal; background-color: #FFFB30; color :black;"> 9) Match win result:</h1>


In [24]:
match_data['result'].value_counts()

wickets    435
runs       364
tie         13
Name: result, dtype: int64

<div class="alert alert-block alert-info"><p style='color:black;'> 📌We can see that <b>435 out of 816 matches was won by team batting second</b> while 364 matches was won by team batting first.
</p>
</div>



<small><a href='#0'>Go to top</a></small>  

<a id="10">
<h1 style = "font-family: Snell Roundhand;font-weight: 500;font-size: 30px; font-style: normal; background-color: #FFFB30; color :black;"> 10) Number of times team have won the tournament :</h1>

In [25]:
winning_teams = match_data[['Season','winner']]

#dictionaries to get winners to each season
winners_team = {}
for i in sorted(winning_teams.Season.unique()):
    winners_team[i] = winning_teams[winning_teams.Season == i]['winner'].tail(1).values[0]
    
winners_of_IPL = pd.Series(winners_team)
winners_of_IPL = pd.DataFrame(winners_of_IPL, columns=['team'])

In [26]:
winners_of_IPL.value_counts().index

MultiIndex([(       'Mumbai Indians',),
            (  'Chennai Super Kings',),
            ('Kolkata Knight Riders',),
            (  'Sunrisers Hyderabad',),
            (     'Rajasthan Royals',),
            (      'Deccan Chargers',)],
           names=['team'])

In [27]:
colors = ['turquoise',] * 6
colors[0] = 'crimson'
fig=px.bar( y=winners_of_IPL['team'].value_counts(),x=winners_of_IPL['team'].value_counts().index,labels=dict(x="Team Name",y="Count"),)
fig.update_layout(title="Winners of IPL",
                  titlefont={'size': 26},template='simple_white'     
                  )
fig.update_traces(marker_line_color='black',
                  marker_line_width=2.5, opacity=1,marker_color=colors)
fig.show()

<div class="alert alert-block alert-info"><p style='color:black;'> 📌
<b>Mumbai Indians had won the IPL 5 times</b> while Chennai Super Kings had won it 3 times.
</p>
</div>

<small><a href='#0'>Go to top</a></small>  

<a id="11">
<h1 style = "font-family: Snell Roundhand;font-weight: 500;font-size: 30px; font-style: normal; background-color: #FFFB30; color :black;"> 11) Total number of matches played by a team :</h1>


In [28]:
matches_played_byteams=pd.concat([match_data['team1'],match_data['team2']],axis=1)
teams=(matches_played_byteams['team1'].value_counts()+matches_played_byteams['team2'].value_counts()).reset_index()
teams.columns=['Team Name','Total Matches played']
teams.sort_values(by=['Total Matches played'],ascending=False).reset_index().drop('index',axis=1).style.background_gradient(cmap='PuBu')

In [29]:
wins=pd.DataFrame(match_data['winner'].value_counts()).reset_index()
wins.columns=['Team Name','Wins']
wins.style.background_gradient(cmap='PuBu')

In [30]:
played=teams.merge(wins,left_on='Team Name',right_on='Team Name',how='inner')
played['% Win']=(played['Wins']/played['Total Matches played'])*100
played.sort_values(by=['% Win'],ascending=False).reset_index().drop('index',axis=1).style.background_gradient(cmap='PuBu',subset=['% Win'])

In [31]:
colors = ['turquoise',] * 15
colors[8] = 'crimson'
fig=px.bar(x=played['Team Name'],y=played['Total Matches played'],labels=dict(x="Team Name",y="Count"),)
fig.update_layout(title="Total number of matches played",
                  titlefont={'size': 26},template='simple_white'     
                  )
fig.update_traces(marker_line_color='black',
                  marker_line_width=2.5, opacity=1,marker_color=colors)
fig.show()

<div class="alert alert-block alert-info"><p style='color:black;'> 📌
Mumbai Indians have played the most matches(203), followed by Royal Challengers Bangalore (195)
</p>
</div>

<small><a href='#0'>Go to top</a></small>  

<a id="12">
<h1 style = "font-family: Snell Roundhand;font-weight: 500;font-size: 30px; font-style: normal; background-color: #FFFB30; color :black;"> 12) Most number of wins :</h1>

In [32]:
colors = ['turquoise',] * 15
colors[8] = 'crimson'
fig=px.bar(x=played['Team Name'],y=played['Wins'],labels=dict(x="Team Name",y="Count"),)
fig.update_layout(title="Total Win by teams",
                  titlefont={'size': 26},template='simple_white'     
                  )
fig.update_traces(marker_line_color='black',
                  marker_line_width=2.5, opacity=1,marker_color=colors)
fig.show()


<div class="alert alert-block alert-info"><p style='color:black;'> 📌
Mumbai Indians had won the most matches(120), followed by Chennai Super Kings (106)
</p>
</div>


<small><a href='#0'>Go to top</a></small>  

<a id="13">
<h1 style = "font-family: Snell Roundhand;font-weight: 500;font-size: 30px; font-style: normal; background-color: #FFFB30; color :black;"> 13) Win % by teams :</h1>


In [33]:
colors = ['turquoise',] * 15
colors[-4] = 'crimson'
fig=px.bar(x=played['Team Name'],y=played['% Win'],labels=dict(x="Team Name",y="Count"),)
fig.update_layout(title="Win % by teams",
                  titlefont={'size': 26},template='simple_white'     
                  )
fig.update_traces(marker_line_color='black',
                  marker_line_width=2.5, opacity=1,marker_color=colors)
fig.show()


<div class="alert alert-block alert-info"><p style='color:black;'> 📌
Rising Pune Supergiants have the highest win % of 62.50, followed by Chennai Super kings and Mumbai Indians.<br> 
This is largely due to the fact that they had played really few matches.
</p>
</div>


<small><a href='#0'>Go to top</a></small>  

<a id="14">
<h1 style = "font-family: Snell Roundhand;font-weight: 500;font-size: 30px; font-style: normal; background-color: #FFFB30; color :black;">14) Lucky Venues for a team :</h1>

In [34]:
def lucky(match_data,team_name):
    return match_data[match_data['winner']==team_name]['venue'].value_counts().nlargest(10)

In [35]:
mi=lucky(match_data,'Mumbai Indians')
values = mi
labels=mi.index
colors = ['turquoise', 'crimson']
fig = go.Figure(data=[go.Pie(labels=labels,values=values,hole=.3)])
fig.update_traces(hoverinfo='label+percent', textinfo='value', textfont_size=20,
                  marker=dict(colors=colors, line=dict(color='#000000', width=3)))
fig.update_layout(title="Wins at different Venues for MI:",
                  titlefont={'size': 30},
                  )
fig.show()

In [36]:
rcb=lucky(match_data,'Royal Challengers Bangalore')
values = rcb
labels=rcb.index
colors = ['turquoise', 'crimson']
fig = go.Figure(data=[go.Pie(labels=labels,values=values,hole=.3)])
fig.update_traces(hoverinfo='label+percent', textinfo='value', textfont_size=20,
                  marker=dict(colors=colors, line=dict(color='#000000', width=3)))
fig.update_layout(title="Wins at different Venues for RCB:",
                  titlefont={'size': 30},
                  )
fig.show()

In [37]:
csk=lucky(match_data,'Chennai Super Kings')
values = csk
labels=csk.index
colors = ['turquoise', 'crimson']
fig = go.Figure(data=[go.Pie(labels=labels,values=values,hole=.3)])
fig.update_traces(hoverinfo='label+percent', textinfo='value', textfont_size=20,
                  marker=dict(colors=colors, line=dict(color='#000000', width=3)))
fig.update_layout(title="Wins at different Venues for CSK:",
                  titlefont={'size': 30},
                  )
fig.show()

<div class="alert alert-block alert-info"><p style='color:black;'> 📌
It can be easily seen that team have won the most of its matches at their home venues.
</p>
</div>



<small><a href='#0'>Go to top</a></small>  

<a id="15">
<h1 style = "font-family: Snell Roundhand;font-weight: 500;font-size: 30px; font-style: normal; background-color: #FFFB30; color :black;">15) Comparision between two teams :</h1>



In [38]:
def comparison(team1,team2):
    compare=match_data[((match_data['team1']==team1)|(match_data['team2']==team1))&((match_data['team1']==team2)|(match_data['team2']==team2))]
    fig=px.histogram(data_frame=compare,x='Season',color='winner',labels=dict(x="Team Name",y="Count"),barmode='group',nbins=16,color_discrete_sequence=colors,)
    fig.update_layout(title="Team Comparision:",
                      titlefont={'size': 26},template='simple_white'     
                      )
    fig.update_traces(marker_line_color='black',
                      marker_line_width=2.5, opacity=1)
    fig.show()
    

In [39]:
comparison('Mumbai Indians','Chennai Super Kings')

<div class="alert alert-block alert-info"><p style='color:black;'> 📌
Mumbai Indians is slight ahead of Chennai Super Kings in this heavy clash.<br>
The dominance was especially seen in the 2019 season, where Mumbai defeated Chennai 4 out of 4 times they met, including the playoff and final.
</p>
</div>

<small><a href='#0'>Go to top</a></small>  

<a id="16">
<h1 style = "font-family: Snell Roundhand;font-weight: 500;font-size: 30px; font-style: normal; background-color: #FFFB30; color :black;">16) Particular batsman analysis :</h1>



<div class="alert alert-block alert-info"><p style='color:black;'> 📌
    Lets see some stats of highest run scorer of IPL : <b>Virat Kohli</b>
</p>
</div>

In [40]:
filt=(deliveries_data['batsman']=='V Kohli')
df_kohli=deliveries_data[filt]
df_kohli.head()

id  inning  ...                 batting_team           bowling_team
211  335982       2  ...  Royal Challengers Bangalore  Kolkata Knight Riders
212  335982       2  ...  Royal Challengers Bangalore  Kolkata Knight Riders
213  335982       2  ...  Royal Challengers Bangalore  Kolkata Knight Riders
216  335982       2  ...  Royal Challengers Bangalore  Kolkata Knight Riders
217  335982       2  ...  Royal Challengers Bangalore  Kolkata Knight Riders

[5 rows x 18 columns]

In [41]:
values = df_kohli['dismissal_kind'].value_counts()
labels=df_kohli['dismissal_kind'].value_counts().index
fig = go.Figure(data=[go.Pie(labels=labels,values=values,hole=.3)])
fig.update_traces(hoverinfo='label+percent', textinfo='label', textfont_size=20,
                  marker=dict(colors=colors, line=dict(color='#000000', width=3)))
fig.update_layout(title="Dismissal Type",
                  titlefont={'size': 30},
                  )
fig.show()

In [42]:
len(df_kohli[df_kohli['batsman_runs']==4])

504

In [43]:
len(df_kohli[df_kohli['batsman_runs']==6])

202

In [44]:
df_kohli['total_runs'].sum()

6081

In [45]:
def count(df_kohli,runs):
    return len(df_kohli[df_kohli['batsman_runs']==runs])*runs

In [46]:
print("Runs scored from 1's :",count(df_kohli,1))
print("Runs scored from 2's :",count(df_kohli,2))
print("Runs scored from 3's :",count(df_kohli,3))
print("Runs scored from 4's :",count(df_kohli,4))
print("Runs scored from 6's :",count(df_kohli,6))

Runs scored from 1's : 1919
Runs scored from 2's : 692
Runs scored from 3's : 39
Runs scored from 4's : 2016
Runs scored from 6's : 1212


In [47]:
values=[1919,692,39,2016,1212]
labels=[1,2,3,4,6]
fig = go.Figure(data=[go.Pie(labels=labels,values=values,hole=.3)])
fig.update_traces(hoverinfo='label+percent', textinfo='label+value', textfont_size=20,
                  marker=dict(colors=colors, line=dict(color='#000000', width=3)))
fig.update_layout(title="Virat Kohli total runs contribution",
                  titlefont={'size': 30},
                  )
fig.show()


<div class="alert alert-block alert-info"><p style='color:black;'> 📌
    Interesting fact is that <b>Virat Kohli</b> has scored more runs by taking singles than by hitting 6
</p>
</div>

<small><a href='#0'>Go to top</a></small>  

<a id="17">
<h1 style = "font-family: Snell Roundhand;font-weight: 500;font-size: 30px; font-style: normal; background-color: #FFFB30; color :black;">17) Innings wise comparision :</h1>



In [48]:
runs=deliveries_data.groupby(['id','inning','batting_team'])[['total_runs']].sum().reset_index()
runs.drop('id',axis=1,inplace=True)
runs.head()

inning                 batting_team  total_runs
0       1        Kolkata Knight Riders         222
1       2  Royal Challengers Bangalore          82
2       1          Chennai Super Kings         240
3       2              Kings XI Punjab         207
4       1             Rajasthan Royals         129

In [49]:
inning1=runs[runs['inning']==1]
inning2=runs[runs['inning']==2]

<p style = "font-size:20px; font-weight : normal; color : #0331A8;">Batting first :</p>





In [50]:
fig = px.box(y='total_runs',x='batting_team',data_frame=inning1,color='batting_team')

fig.update_layout(title="Batting First",
                  titlefont={'size': 26},template='simple_white'
                  )
fig.show()

<div class="alert alert-block alert-info"><p style='color:black;'> 📌
Royal Challengers Bangalore and Mumbai Indians median value is better than other teams while batting first.<br>
Royal Challengers Bangalore had scored 250+ in a single match and is the only team to achieve that feat.
</p>
</div>



<p style = "font-size:20px; font-weight : normal; color : #0331A8;">Batting second :</p>




In [51]:
fig = px.box(y='total_runs',x='batting_team',data_frame=inning2,color='batting_team')

fig.update_layout(title="Batting Second",
                  titlefont={'size': 26},template='simple_white'
                  )
fig.show()

<small><a href='#0'>Go to top</a></small>  

<a id="18">
<h1 style = "font-family: Snell Roundhand;font-weight: 500;font-size: 30px; font-style: normal; background-color: #FFFB30; color :black;">18) Scored 200+ runs :</h1>


In [52]:
high_scores=deliveries_data.groupby(['id', 'inning','batting_team','bowling_team'])['total_runs'].sum().reset_index() 
score_200=high_scores[high_scores['total_runs']>=200]
score_200.head(5)

id  inning  ...                 bowling_team total_runs
0   335982       1  ...  Royal Challengers Bangalore        222
2   335983       1  ...              Kings XI Punjab        240
3   335983       2  ...          Chennai Super Kings        207
14  335989       1  ...               Mumbai Indians        208
15  335989       2  ...          Chennai Super Kings        202

[5 rows x 5 columns]

In [53]:
x1=score_200['batting_team'].value_counts()
x1=pd.DataFrame(x1)
x1.style.background_gradient(cmap='PuBu')

In [54]:
colors = ['turquoise',] * 11
colors[0] = 'crimson'
fig=px.bar(x=x1.index,y=x1['batting_team'],labels=dict(x="Team Name",y="Count"),)
fig.update_layout(title="Total count of 200+ by batting team",
                  titlefont={'size': 26},template='simple_white'     
                  )
fig.update_traces(marker_line_color='black',
                  marker_line_width=2.5, opacity=1,marker_color=colors)
fig.show()

<div class="alert alert-block alert-info"><p style='color:black;'> 📌
Royal Challengers Bangalore had scored the most 200+ score (18 times), followed by Chennai Super Kings who had scored 17 times.
</p>
</div>


<small><a href='#0'>Go to top</a></small>  

<a id="19">
<h1 style = "font-family: Snell Roundhand;font-weight: 500;font-size: 30px; font-style: normal; background-color: #FFFB30; color :black;">19) Conceded 200+ runs :</h1>


In [55]:
z=score_200['bowling_team'].value_counts()
z=pd.DataFrame(z)
z.style.background_gradient(cmap='PuBu')


In [56]:
colors = ['turquoise',] * 13
colors[0] = 'crimson'
fig=px.bar(x=z.index,y=z['bowling_team'],labels=dict(x="Team Name",y="Count"),)
fig.update_layout(title="Total count of 200+ conceded by bowling team",
                  titlefont={'size': 26},template='simple_white'     
                  )
fig.update_traces(marker_line_color='black',
                  marker_line_width=2.5, opacity=1,marker_color=colors)
fig.show()


<div class="alert alert-block alert-info"><p style='color:black;'> 📌
Kings XI Punjab has conceded 200+ runs 20 times, followed by Royal Challengers Bangalore (17)
</p>
</div>

<small><a href='#0'>Go to top</a></small>  

<a id="20">
<h1 style = "font-family: Snell Roundhand;font-weight: 500;font-size: 30px; font-style: normal; background-color: #FFFB30; color :black;">20) Highest runs in an innings :</h1>



In [57]:
high_200=deliveries_data.groupby(['id', 'inning','batting_team','bowling_team'])['total_runs'].sum().reset_index()
high_200.set_index(['id'],inplace=True)

In [58]:
high_200['total_runs'].max()

263

<div class="alert alert-block alert-info"><p style='color:black;'> 📌
In season 2013, Royal Challengers Bangalore scored 263/5 against Pune Warriors India.
</p>
</div>

<small><a href='#0'>Go to top</a></small>  

<a id="21">
<h1 style = "font-family: Snell Roundhand;font-weight: 500;font-size: 30px; font-style: normal; background-color: #FFFB30; color :black;">21) Biggest win in terms of run margin :</h1>


In [59]:
match_data[match_data['result_margin']==match_data['result_margin'].max()]

id   city        date  ...    umpire2 Season  toss_win_game_win
620  1082635  Delhi  2017-05-06  ...  CK Nandan   2017                 No

[1 rows x 19 columns]

<div class="alert alert-block alert-info"><p style='color:black;'> 📌
In season 2017, Mumbai Indians had defeated Delhi Daredevils by a huge margin of 146 runs.
</p>
</div>


<small><a href='#0'>Go to top</a></small>  

<a id="22">
<h1 style = "font-family: Snell Roundhand;font-weight: 500;font-size: 30px; font-style: normal; background-color: #FFFB30; color :black;">22) Most balls played by a batsman:</h1>


In [60]:
balls_played=deliveries_data.groupby(['batsman'])['ball'].count().reset_index()
balls_played.sort_values(by='ball',ascending=False).head(10).style.background_gradient(cmap='PuBu')


<div class="alert alert-block alert-info"><p style='color:black;'> 📌
Virat kohli had played the most balls.<br> 
David Warner and Chris Gayle are the only two foreign players in this top 10 list.
</p>
</div>

<small><a href='#0'>Go to top</a></small>  

<a id="23">
<h1 style = "font-family: Snell Roundhand;font-weight: 500;font-size: 30px; font-style: normal; background-color: #FFFB30; color :black;">23) Top 10 run scorer of all time :</h1>


In [61]:
runs=deliveries_data.groupby(['batsman'])['batsman_runs'].sum().reset_index()
runs.columns=['Batsman','runs']
y=runs.sort_values(by='runs',ascending=False).head(10).reset_index().drop('index',axis=1)
y.style.background_gradient(cmap='PuBu')

In [62]:
colors = ['turquoise',] * 13
colors[0] = 'crimson'
fig=px.bar(x=y['Batsman'],y=y['runs'],labels=dict(x="Player",y="Total Runs"),)
fig.update_layout(title="Top 10 leading run-scrorer",
                  titlefont={'size': 26},template='simple_white'     
                  )
fig.update_traces(marker_line_color='black',
                  marker_line_width=2.5, opacity=1,marker_color=colors)
fig.show()


<div class="alert alert-block alert-info"><p style='color:black;'> 📌
Virat Kohli is the leading run scorer in IPL.<br>
One interesting thing to notice that MS Dhoni is the only player in this list who bats down the order.
</p>
</div>

<small><a href='#0'>Go to top</a></small>  

<a id="24">
<h1 style = "font-family: Snell Roundhand;font-weight: 500;font-size: 30px; font-style: normal; background-color: #FFFB30; color :black;">24) Most number of 4's :</h1>


In [63]:
balls_played=balls_played.merge(runs,left_on='batsman',right_on='Batsman',how='outer')
four=deliveries_data[deliveries_data['batsman_runs']==4]
runs_4=four.groupby('batsman')['batsman_runs'].count().reset_index()
runs_4.columns=['Batsman','4s']
runs_4.sort_values(by='4s',ascending=False).head(10).reset_index().drop('index',axis=1).style.background_gradient(cmap='PuBu')

<div class="alert alert-block alert-info"><p style='color:black;'> 📌
Shikhar Dhawan holds the record for most number of 4's
</p>
</div>

<small><a href='#0'>Go to top</a></small>  

<a id="25">
<h1 style = "font-family: Snell Roundhand;font-weight: 500;font-size: 30px; font-style: normal; background-color: #FFFB30; color :black;">25) Most number of 6's :</h1>


In [64]:
six=deliveries_data.groupby('batsman')['batsman_runs'].agg(lambda x: (x==6).sum()).reset_index()
six.columns=['Batsman','6s']
six.sort_values(by='6s',ascending=False).head(10).reset_index().drop('index',axis=1).style.background_gradient(cmap='PuBu')

<div class="alert alert-block alert-info"><p style='color:black;'> 📌
"The Universe Boss" Chris Gayle had smashed 349 sixes, most by any batsman.
</p>
</div>

<small><a href='#0'>Go to top</a></small>  

<a id="26">
<h1 style = "font-family: Snell Roundhand;font-weight: 500;font-size: 30px; font-style: normal; background-color: #FFFB30; color :black;">26) Highest Strike rate (minimum 100 balls):</h1>


In [65]:
player=pd.concat([runs,balls_played.iloc[:,1],runs_4.iloc[:,1],six.iloc[:,1]],axis=1)
player['strike_rate']=player['runs']/player['ball']*100
player['4s'].fillna(0,inplace=True)
player.isnull().values.any()

False

In [66]:
player.sort_values(by='strike_rate',ascending=False).head(10)

Batsman  runs  ball     4s   6s  strike_rate
72        B Stanlake     5     2    3.0    0   250.000000
504         Umar Gul    39    19    0.0    5   205.263158
395         RS Sodhi     4     2    1.0    0   200.000000
470    Shahid Afridi    81    46    0.0    6   176.086957
175       I Malhotra     7     4   13.0    0   175.000000
498     TU Deshpande    21    12    0.0    1   175.000000
33        AD Russell  1517   882   24.0  129   171.995465
253        LJ Wright   106    63    4.0    3   168.253968
57       Abdul Samad   111    66   28.0    6   168.181818
235  KMDN Kulasekara     5     3  156.0    0   166.666667

In [67]:
sr=player[player.ball > 100]
sr.sort_values(by='strike_rate',ascending=False).head(10)


Batsman  runs  ball     4s   6s  strike_rate
33      AD Russell  1517   882   24.0  129   171.995465
217      K Gowtham   186   113  247.0   12   164.601770
80     BCJ Cutting   238   146   40.0   19   163.013699
317       N Pooran   521   323    6.0   39   161.300310
453      SP Narine   892   573    0.0   52   155.671902
293         MM Ali   309   199   23.0   23   155.276382
97       CH Morris   551   360  116.0   30   153.055556
192      JC Archer   195   128   18.0   14   152.343750
106  CR Brathwaite   181   120   11.0   16   150.833333
88    Bipul Sharma   187   124    5.0    9   150.806452

In [68]:
reqsr=sr.drop(columns=['runs','ball','4s','6s'],axis=1)
reqsr.sort_values(by='strike_rate',ascending=False).head(10).style.background_gradient(cmap='PuBu')

<div class="alert alert-block alert-info"><p style='color:black;'> 📌
Andre Russel has the highest strike rate, followed by K. Gowtham and B. Cutting.
</p>
</div>

<small><a href='#0'>Go to top</a></small>  

<a id="27">
<h1 style = "font-family: Snell Roundhand;font-weight: 500;font-size: 30px; font-style: normal; background-color: #FFFB30; color :black;">27) Highest wicket-taker :</h1>


In [69]:
deliveries_data['dismissal_kind'].unique()
dismissal_kinds = ['caught', 'bowled', 'lbw', 'caught and bowled',
       'stumped', 'hit wicket']
hwt=deliveries_data[deliveries_data["dismissal_kind"].isin(dismissal_kinds)]
bo=hwt['bowler'].value_counts()

In [70]:
colors = ['turquoise',] * 13
colors[0] = 'crimson'
fig=px.bar(x=bo[:10].index,y=bo[:10],labels=dict(x="Bowler",y="Total Wickets"),)
fig.update_layout(title="Leading wicket-takers",
                  titlefont={'size': 26},template='simple_white'     
                  )
fig.update_traces(marker_line_color='black',
                  marker_line_width=2.5, opacity=1,marker_color=colors)
fig.show()


<div class="alert alert-block alert-info"><p style='color:black;'> 📌
L. Malinga is the leading wicket taker in IPL.<br>
6 out of top 10 are spin bowlers.
</p>
</div>

<small><a href='#0'>Go to top</a></small>  

<a id="28">
<h1 style = "font-family: Snell Roundhand;font-weight: 500;font-size: 30px; font-style: normal; background-color: #FFFB30; color :black;">28) Total count of matches played in differnt stadiums :</h1>



In [71]:
colors = ['turquoise',] * 13
colors[0] = 'crimson'
fig=px.bar(x=match_data['venue'].value_counts()[:10].index,y=match_data['venue'].value_counts()[:10],labels=dict(x="Venue",y="Match count"),)
fig.update_layout(title="Matches played at different stadiums",
                  titlefont={'size': 26},template='simple_white'     
                  )
fig.update_traces(marker_line_color='black',
                  marker_line_width=2.5, opacity=1,marker_color=colors)
fig.show()



<div class="alert alert-block alert-info"><p style='color:black;'> 📌
Eden Gardens has hosted most number of matches.
</p>
</div>


<small><a href='#0'>Go to top</a></small>  

<a id="29">
<h1 style = "font-family: Snell Roundhand;font-weight: 500;font-size: 30px; font-style: normal; background-color: #FFFB30; color :black;">29) Man of the Match award :</h1>


In [72]:
colors = ['turquoise',] * 11
colors[0] = 'crimson'
fig=px.bar(x=match_data.player_of_match.value_counts()[:10].index,y=match_data.player_of_match.value_counts()[:10],labels=dict(x="Players",y="Count"),)
fig.update_layout(title="Top 10 MOM awardee",
                  titlefont={'size': 26},template='simple_white'     
                  )
fig.update_traces(marker_line_color='black',
                  marker_line_width=2.5, opacity=1,marker_color=colors)
fig.show()

<div class="alert alert-block alert-info"><p style='color:black;'> 📌
<b>AB de Villiers</b> (23) had won the most MOM awards, followed by Chris Gayle (22)
</p>
</div>


<small><a href='#0'>Go to top</a></small>  

<a id="30">
<h1 style = "font-family: Snell Roundhand;font-weight: 500;font-size: 30px; font-style: normal; background-color: #FFFB30; color :black;">30) Total number of fours in each season :</h1>



In [73]:
data_4 = match_data['Season'].unique()

fours_list = []
for var in data_4:
    new_df = match_data[match_data['Season']==var]
    total_fours = 0
    for i in new_df['id'].values:
        temp_df = deliveries_data[deliveries_data['id']==i]
        fours = temp_df[temp_df['batsman_runs']==4]['batsman_runs'].count()
        total_fours+=fours
    fours_list.append(total_fours)
    

colors = ['turquoise',] * 14
colors[5] = 'crimson'
fig=px.bar(x=data_4, y=fours_list,labels=dict(x="Season",y="Total Fours"),)
fig.update_layout(title="Total number of Fours in each season",
                  titlefont={'size': 26},template='simple_white'     
                  )
fig.update_traces(marker_line_color='black',
                  marker_line_width=2.5, opacity=1,marker_color=colors)
fig.show()


<div class="alert alert-block alert-info"><p style='color:black;'> 📌
Highest number of four in a season was is 2013 season while season 2009 had the lowest count of 4's.
</p>
</div>


<small><a href='#0'>Go to top</a></small>  

<a id="31">
<h1 style = "font-family: Snell Roundhand;font-weight: 500;font-size: 30px; font-style: normal; background-color: #FFFB30; color :black;">31) Total number of sixes in each season :</h1>


In [74]:

data_6 = match_data['Season'].unique()

# Getting total sixes from each season by check id from matches dataset in deliveries dataset.
sixes_list = []
for var in data_6:
    new_df = match_data[match_data['Season']==var]
    total_sixes = 0
    for i in new_df['id'].values:
        temp_df = deliveries_data[deliveries_data['id']==i]
        sixes = temp_df[temp_df['batsman_runs']==6]['batsman_runs'].count()
        total_sixes+=sixes
    sixes_list.append(total_sixes)
    
colors = ['turquoise',] * 14
colors[-4] = 'crimson'
fig=px.bar(x=data_4, y=sixes_list,labels=dict(x="Season",y="Total Sixes"),)
fig.update_layout(title="Total number of Sixes in each season",
                  titlefont={'size': 26},template='simple_white'     
                  )
fig.update_traces(marker_line_color='black',
                  marker_line_width=2.5, opacity=1,marker_color=colors)
fig.show()


<div class="alert alert-block alert-info"><p style='color:black;'> 📌
In season 2018, the maximum number of sixes were hit while the lowest was observed in season 2009.
</p>
</div>

<small><a href='#0'>Go to top</a></small>  

<a id="32">
<h1 style = "font-family: Snell Roundhand;font-weight: 500;font-size: 30px; font-style: normal; background-color: #FFFB30; color :black;">32) Total runs scored from boundaries in each season :</h1>


In [75]:
runs4=np.dot(fours_list,4)
runs6=np.dot(sixes_list,6)

In [76]:
k=runs4+runs6
Y=match_data['Season'].unique()
colors = ['turquoise',] * 14
colors[5] = 'crimson'
fig=px.bar(x=Y,y=k,labels=dict(x="Season",y="Total runs"),)
fig.update_layout(title="Total number of runs scored from boundaries in each season",
                  titlefont={'size': 26},template='simple_white'     
                  )
fig.update_traces(marker_line_color='black',
                  marker_line_width=2.5, opacity=1,marker_color=colors)
fig.show()



<div class="alert alert-block alert-info"><p style='color:black;'> 📌
Total run scored from boundaries is lowest in season 2009 and highest in season 2013.
</p>
</div>


<small><a href='#0'>Go to top</a></small>  

<a id="33">
<h1 style = "font-family: Snell Roundhand;font-weight: 500;font-size: 30px; font-style: normal; background-color: #FFFB30; color :black;">33) Total contribution of runs from boundaries in each season :</h1>


In [77]:
totruns=np.array(season['total_runs'])
res=(k/totruns)*100

In [78]:
res

array([58.81697051, 50.88235294, 54.82400339, 54.35378652, 53.63203135,
       54.3809059 , 55.6983447 , 57.71328824, 54.95705652, 56.87037136,
       59.49449776, 58.32989691, 55.50847458])

In [79]:
colors = ['turquoise',] * 14
colors[10] = 'crimson'
fig=px.bar(x=Y,y=res,labels=dict(x="Season",y="Percentage"),)
fig.update_layout(title="Total contribution of runs from boundaries in each season",
                  titlefont={'size': 26},template='simple_white'     
                  )
fig.update_traces(marker_line_color='black',
                  marker_line_width=2.5, opacity=1,marker_color=colors)
fig.show()


<div class="alert alert-block alert-info"><p style='color:black;'> 📌
In season 2018, 59.49 % runs of the total runs came from boundaries while 50.88 % runs came from boundaries in season 2009 which is lowest till now.
</p>
</div>


<small><a href='#0'>Go to top</a></small>  

<a id="34">
<h1 style = "font-family: Snell Roundhand;font-weight: 500;font-size: 30px; font-style: normal; background-color: #FFFB30; color :black;">34) Total runs scored by teams in first 6 overs :</h1>



In [80]:
team = deliveries_data['batting_team'].unique()
team_runs = []
for var in team:
    temp_df = deliveries_data[deliveries_data['batting_team']==var]
    temp_df = temp_df[temp_df['over'].isin([0,1,2,3,4,5])]
    runs = temp_df['total_runs'].sum()
    team_runs.append(runs)
team = pd.DataFrame(data=team_runs, index=team,columns=['Runs In First 6 Overs'])
#team.sort_values('Runs In First 6 Overs', ascending=False, inplace=True)
team.index.name = 'Team'

colors = ['turquoise',] * 15
colors[6] = 'crimson'
fig=px.bar(x=team.index,y=team['Runs In First 6 Overs'],labels=dict(x="Team Name",y="Total runs"),)
fig.update_layout(title="Total runs scored by teams in thier first 6 overs",
                  titlefont={'size': 26},template='simple_white'     
                  )
fig.update_traces(marker_line_color='black',
                  marker_line_width=2.5, opacity=1,marker_color=colors)
fig.show()


<div class="alert alert-block alert-info"><p style='color:black;'> 📌
<b>Mumbai Indians</b> had scored the  most runs in first 6 overs, followed by Kings XI Punjab.
</p>
</div>


<small><a href='#0'>Go to top</a></small>  

<a id="35">
<h1 style = "font-family: Snell Roundhand;font-weight: 500;font-size: 30px; font-style: normal; background-color: #FFFB30; color :black;">35) Total runs scored by teams in last 4 overs :</h1>



In [81]:
team1 = deliveries_data['batting_team'].unique()
team_runs1 = []
for var in team1:
    temp_df = deliveries_data[deliveries_data['batting_team']==var]
    temp_df = temp_df[temp_df['over'].isin([19,18,17,16])]
    runs1 = temp_df['total_runs'].sum()
    team_runs1.append(runs1)
team1 = pd.DataFrame(data=team_runs1, index=team1,columns=['Runs In Last 4 Overs'])
# team.sort_values('Runs In First 6 Overs', ascending=False, inplace=True)
team1.index.name = 'Team'
colors = ['turquoise',] * 15
colors[6] = 'crimson'
fig=px.bar(x=team1.index,y=team1['Runs In Last 4 Overs'],labels=dict(x="Team Name",y="Total runs"),)
fig.update_layout(title="Total runs scored by teams in thier last 4 overs",
                  titlefont={'size': 26},template='simple_white'     
                  )
fig.update_traces(marker_line_color='black',
                  marker_line_width=2.5, opacity=1,marker_color=colors)
fig.show()


<div class="alert alert-block alert-info"><p style='color:black;'> 📌
Mumbai Indians had scored most runs in last 4 overs, followed by Royal Challengers Bangalore.
</p>
</div>

<small><a href='#0'>Go to top</a></small>  

<a id="36">
<h1 style = "font-family: Snell Roundhand;font-weight: 500;font-size: 30px; font-style: normal; background-color: #FFFB30; color :black;">36) Highest scoring run-rate in first 6 overs :</h1>



In [82]:
played1=played.merge(team,left_on='Team Name',right_on='Team',how='inner')
played3=played1.merge(team1,left_on='Team Name',right_on='Team',how='inner')
mintoover=np.dot(played3['Total Matches played'],6)
mintoover1=np.dot(played3['Total Matches played'],4)
played3['RR in first 6 overs']=(played3['Runs In First 6 Overs']/mintoover)
played3['RR in last 4 overs']=(played3['Runs In Last 4 Overs']/mintoover1)
played3

Team Name  ...  RR in last 4 overs
0           Chennai Super Kings  ...            9.346910
1               Deccan Chargers  ...            8.463333
2                Delhi Capitals  ...            8.787879
3              Delhi Daredevils  ...            7.830745
4                 Gujarat Lions  ...            7.675000
5               Kings XI Punjab  ...            8.193421
6          Kochi Tuskers Kerala  ...            6.017857
7         Kolkata Knight Riders  ...            8.104167
8                Mumbai Indians  ...            9.283251
9                 Pune Warriors  ...            7.391304
10             Rajasthan Royals  ...            7.830745
11       Rising Pune Supergiant  ...            8.671875
12      Rising Pune Supergiants  ...            7.910714
13  Royal Challengers Bangalore  ...            8.701282
14          Sunrisers Hyderabad  ...            8.377016

[15 rows x 8 columns]

In [83]:
colors = ['turquoise',] * 15
colors[4] = 'crimson'
fig=px.bar(x=played3['Team Name'],y=played3['RR in first 6 overs'],labels=dict(x="Team Name",y="Run Rate"),)
fig.update_layout(title="Run Rate in first 6 overs",
                  titlefont={'size': 26},template='simple_white'     
                  )
fig.update_traces(marker_line_color='black',
                  marker_line_width=2.5, opacity=1,marker_color=colors)
fig.show()


<div class="alert alert-block alert-info"><p style='color:black;'> 📌
Gujarat Lions has the highest scoring run-rate in first 6 overs, followed by Rising Pune Supergiant.
</p>
</div>


<a id="37">
<h1 style = "font-family: Snell Roundhand;font-weight: 500;font-size: 30px; font-style: normal; background-color: #FFFB30; color :black;">37) Highest scoring rate in last 4 overs :</h1>

In [84]:
colors = ['turquoise',] * 15
colors[0] = 'crimson'
fig=px.bar(x=played3['Team Name'],y=played3['RR in last 4 overs'],labels=dict(x="Team Name",y="Run Rate"),)
fig.update_layout(title="Run Rate in last 4 overs",
                  titlefont={'size': 26},template='simple_white'     
                  )
fig.update_traces(marker_line_color='black',
                  marker_line_width=2.5, opacity=1,marker_color=colors)
fig.show()


<div class="alert alert-block alert-info"><p style='color:black;'> 📌
<b>Chennai Super Kings</b> has the highest scoring run-rate in last 4 overs, followed by Mumbai Indians.
</p>
</div>
